### PR TITLE
WIP: switch to Tox + pyup.io

### PR DESCRIPTION
**Pull request by stephendonner**
Caveats:
1) Very much a work in progress.
2) Doesn't run; error output here: https://pastebin.mozilla.org/8957146
3) I'm doing a bunch of these Tox/pyup.io PRs today, so apologies in advance; I'm very likely to have missed something

Starts to address the following:

* Switch to Tox: https://github.com/mozilla/mozwebqa-examples/issues/7
* Enable pyup.io: https://github.com/mozilla/mozwebqa-examples/issues/8
* Fix/populate empty README: https://github.com/mozilla/mozwebqa-examples/issues/2

@davehunt feedback/r? highly welcome!

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,10 +5,11 @@ python:
 before_install:
   - "export DISPLAY=:99.0"
   - "sh -e /etc/init.d/xvfb start"
+env:
+  - TOXENV=flake8
```
Comment (davehunt) [inline .travis.yml:10]: This will cause only the `flake8` environment to run in Travis CI.

File: requirements/tests.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Flask
```
Comment (davehunt) [inline requirements/tests.txt]: Flask is a requirement for the application under test. All other requirements are for the tests themselves. I would create a `base.txt` for the application requirements, and make `tests.txt` include these.

File: setup.cfg [hunk 3]
```diff
@@ -1,2 +1,8 @@
 [flake8]
 ignore=E501
+
+[tool:pytest]
+python_files=test_*.py
```
Comment (davehunt) [inline setup.cfg]: I believe this is the default, so shouldn't be needed.

File: setup.cfg [hunk 4]
```diff
@@ -1,2 +1,8 @@
...
+addopts=-n=auto --verbose -r=a --junit-xml=results/junit.xml
```
Comment (davehunt) [inline setup.cfg]: We're not currently running these tests in parallel in Travis CI, so I wonder if we should avoid introducing this (we should probably remove the pytest-xdist dependency too, if there's an issue with running them in parallel).

File: tox.ini [hunk 5]
```diff
@@ -0,0 +1,13 @@
+[tox]
+skipsdist = true
+envlist = tests, flake8
```
Comment (davehunt) [inline tox.ini:3]: I think we should run these tests in multiple Python versions, but that can be a later enhancement.

File: tox.ini [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+[testenv]
+passenv = PYTEST_ADDOPTS
+deps = -rrequirements/tests.txt
+commands = py.test {posargs}
```
Comment (davehunt) [inline tox.ini]: Nit: Remove the dot from `py.test`

File: .travis.yml [hunk 7]
```diff
@@ -5,10 +5,11 @@ python:
...
 install:
-  - "pip install flake8"
-  - "pip install -Ur requirements.txt"
-before_script: "flake8 ."
-script: "py.test -v --driver Firefox --variables variables.json tests"
```
Comment (davehunt) [inline .travis.yml:12]: We still need to pass the variables file, though we can probably just do this in `tox.ini` as this file is in tree and contains no real secrets.

**Review by davehunt — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 7 above)
Comment (stephendonner) [reply .travis.yml:12]: I gave this a stab - I /think/ it's actually being passed in, but as there are new failures (hopefully unrelated and are rather from this repo last being touched 8 months ago), it's hard for me to tell.

File: tox.ini (hunk 5 above)
Comment (stephendonner) [reply tox.ini:3]: +1 - I volunteer to help with a sweep of our projects, at some point in the future, given priorities + time :-)

**Comment by stephendonner** [comment]
@davehunt feedback/r?  - definitely need help understanding and trying to address the actual test failures, now, which we'll need, probably, for this PR to be merged.  Thanks as always!

File: setup.cfg [hunk 8]
```diff
@@ -1,2 +1,7 @@
 [flake8]
 ignore=E501
+
+[tool:pytest]
+addopts = --variables=variables.json --verbose -r=a --junit-xml=results/junit.xml
```
Comment (davehunt) [inline setup.cfg:5]: This approach should work, but I'd at least put `--variables` and `--junit-xml` directly into `tox.ini` in the `commands` line. You're also missing the `--driver` default value.

**Review by davehunt — CHANGES_REQUESTED** [review]
This is failing for Python 2.7 because it's depending on a private method from pytest-selenium that was moved to pytest-base-url. You could resolve this by pinning to pytest-selenium 1.2.1, or by changing lines 36-37 of `conftest.py` to:

```python
from pytest_base_url import plugin
plugin._verify_url(request, base_url)
```

I think we can safely drop support for Python 2.6, so let's do that and ignore those failures.

**Review by davehunt — CHANGES_REQUESTED** [review]
Thinking about this some more, I am not sure it's worth the effort getting this repository into a good state. I think it would be more useful to point contributors to https://github.com/davehunt/cookiecutter-pytest-selenium, which creates a local project with tests that use our recommended practices and tools. What do you think @stephendonner?

**Comment by stephendonner** [comment]
@davehunt yes, I'd be fine (happy, even) with just focusing our efforts on pointing people to (and helping file/fix Issues) cookiecutter-pytest-selenium.  Should we now work on decommissioning this repo with a notice, etc.?  And, if so, just leave as-is, (code-wise), but change the readme?

**Comment by davehunt** [comment]
I doubt anyone is using this repository, but let's do what we did with moz-grid-config and note in the README that we're no longer using or maintaining it. https://github.com/mozilla/moz-grid-config

**Comment by stephendonner** [comment]
imma close this and since we decided to just update the README with a notice, next up is a PR for that, separately.

**Pull request closed without merging**